### ISSUE TITLE
Question: simple example poor performance, what am I doing wrong?

### ISSUE THREAD

**Issue by mfouesneau** [google/neural-tangents#163]
Dear team, great package, I'm very excited to use it. 

However, I tried a simple case, and I failed miserably to get a decent performance.

I generate a multi-dimensional dataset with a relatively simple feature
```python
import numpy as np
    
#Create some fake data
np.random.seed(0)
m = 1000
n = 10
noise_std = 1.
X = 80*numpy.random.uniform(size=(m,n)) - 40
y = np.abs(X[:,6] - 4.0) + noise_std * np.random.normal(size=m)
```    

And I followed your examples as

```python
import neural_tangents as nt
from neural_tangents import stax
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test = train_test_split(
    X, y.reshape(-1, 1), test_size=0.4, random_state=42)

init_fn, apply_fn, kernel_fn = stax.serial(
    stax.Dense(256), stax.Relu(),
    stax.Dense(1)
)
predict_fn = nt.predict.gradient_descent_mse_ensemble(
    kernel_fn, 
    x_train,
    y_train)

# Unpack the predictions namedtuple
y_test_nngp, y_test_ntk = predict_fn(x_test=x_test, get=('nngp', 'ntk'), compute_cov=True)
```

Visual inspection shows terrible predictions, and loss values are large:
```python
loss = lambda ypred, y_hat: 0.5 * jnp.mean((ypred - y_hat) ** 2)
print("loss_nngp = {}".format(loss(y_test_nngp.mean, y_test)))
print("loss_ntk = {}".format(loss(y_test_ntk.mean, y_test)))
```
```
loss_nngp = 6.877374649047852
loss_ntk = 6.610106468200684
```

I varied the network in many ways and fiddled with `learning_rate`  and `diag_reg`, but I hardly changed anything.

I'm sure I am doing something wrong, but I cannot see what it is. Any obvious mistake?

Thanks for your help.

**Comment by romanngg** [comment]
At a glance, library usage seems good to me! Perhaps one way to figure this out is to establish a baseline using some other method (kernel, neural network, etc), to figure out what loss values are expected? For example it seems that `y` will have a mean of 20 (expectation of the absolute value of a uniform from -44 to 36 is 1/2 * (40 + 4) / 2 + 1/2 * (40 - 4) / 2), so scale of outputs is pretty large, so it's not obvious to me if the loss values are that large. Another angle is to try increasing the training set size - it's hard to say if 600 training points is large enough for the model to learn well.

**Comment by mfouesneau** [comment]
![image](https://user-images.githubusercontent.com/1703988/186459332-c67a0d0b-a011-4fb6-8af5-fa31a535592d.png)

The thing is that changing the layer from 50 nodes to 5000 hardly changes the output. I would expect at least some changes.

I tried 10_000 points, and I only gained a factor of 2 on the loss
![image](https://user-images.githubusercontent.com/1703988/186460083-01c8bf76-9419-487e-9d7e-858c692468a1.png)

Is there any guidance on what a correct training set should be?

**Comment by romanngg** [comment]
Note that in your example you are doing inference with an infinitely-wide neural network (`kernel_fn`), so the width doesn't matter in this case. Also, the plot does look like the learned function mimicks `|x_6 - 4|` (at least it's not doing something obviously wrong, it has the right shape and kink location), so I'm inclined to think that it's working as intended?... 

Re training set, I think it's constructed correctly, I'm just not sure how to reason about the generalization that we should expect from it (per your plot, it seems to be at least OKish?...).

And yes, 100K is too much for most GPUs.

**Comment by mfouesneau** [comment]
You're right; it seems to be doing ok, but with serious overfit.

Is there a paper to read to get a feeling for appropriate network architecture? My understanding is that multiplying layers will not change anything unless a "layer" is a complex thing already. right?

![image](https://user-images.githubusercontent.com/1703988/186617289-2340200b-1aca-41ad-9a2d-2da1cde0269c.png)
![image](https://user-images.githubusercontent.com/1703988/186617445-0a8bb898-9c79-4ce2-8c99-9472e8bce58b.png)
